backend/controllers/user.controller.ts

```diff
@@ -65,155 +51,101 @@ export const createNewUser = async (
-    });
-  } catch (error) {
-    next(error);
-  }
-};
-
-// send new OTP
-export const resendOTP = async (
-  req: Request,
-  res: Response,
-  next: NextFunction,
-) => {
-  try {
-    const { email } = otpRequestSchema.parse(req.body);
-
-    const otp = otpGenerator.generate(6, {
-      upperCaseAlphabets: false,
-      lowerCaseAlphabets: false,
-      specialChars: false,
-    });
-
-    await prisma.user.update({
-      where: {
-        email: email,
+      process.env.JWT_SECRET ?? "testSecret",
+      {
+        expiresIn: "1h",
+        mutatePayload: false,
       },
-      data: {
-        otp: otp,
-      },
-    });
+    );
 
     const message = {
       from: "nomansanjari2001@gmail.com",
       to: email,
-      subject: "Verify OTP - CampusBuddy",
-      html: `<b>${otp}</b>`,
+      subject: "Verify your account - CampusBuddy",
+      html: `Verify your account by clicking the link!<br>
+      <a href="http://localhost:3000/api/user/verifyAccount/${token}></a>"`,
     };
 
     await transporter.sendMail(message);
 
     res.status(200).json({
       success: true,
-      message: "New OTP sent",
+      message: "User created, awaiting verification",
     });
   } catch (error) {
     next(error);
   }
 };
 
 // verify OTP
-export const verifyOTP = async (
+export const verifyAccount = async (
   req: Request,
   res: Response,
   next: NextFunction,
 ) => {
   try {
-    const { email, otp } = otpVerifySchema.parse(req.body);
-
-    const user = await prisma.user.findUnique({
-      where: {
-        email: email,
-        otp: otp,
-      },
-    });
+    const token = req.params.token;
 
-    if (user) {
-      const token = jwt.sign(
-        { ID: user.id },
+    try {
+      const payload: string | JwtPayload = jwt.verify(
+        token,
         process.env.JWT_SECRET ?? "testSecret",
       );
 
-      await prisma.user.update({
-        where: {
-          id: user.id,
-        },
-        data: {
-          jwt: token,
-          isVerified: false,
-        },
-      });
+      if (
+        typeof payload === "object" &&
+        payload !== null &&
+        "username" in payload &&
+        "firstName" in payload &&
+        "lastName" in payload &&
+        "email" in payload &&
+        "password" in payload &&
+        "institutionID" in payload
+      ) {
+        const institution = await prisma.institution.findUnique({
+          where: {
+            id: payload.institutionID,
+          },
+        });
 
-      res.status(200).json({
-        success: true,
-        message: "User verified",
-      });
-    } else {
+        if (institution) {
+          const newUser = await prisma.user.create({
+            data: {
+              username: payload.username,
+              firstName: payload.firstName,
+              lastName: payload.lastName,
+              email: payload.email,
+              password: payload.password,
+              institutionId: payload.institutionID,
```

Comment:
Reflect same requested changes above. Keep it consistent and use `institutionId`

Reply:
all instances of institutionID have been changed to institutionId